core/src/proto/types.rs

```diff
@@ -1511,6 +1536,132 @@ impl Tx {
         }
         filtered
     }
+
+    /// Add an extra section to the tx builder by hash
+    pub fn add_extra_section_from_hash(
+        &mut self,
+        hash: crate::types::hash::Hash,
+    ) -> (&mut Self, crate::types::hash::Hash) {
```

Comment:
I think we could just return the hash from these for same reason as above and we cannot use these to chain calls  on `Tx` because of the tuple anyway